document/core/exec/instructions.rst

```diff
@@ -182,6 +182,540 @@ Where the underlying operators are non-deterministic, because they may return on
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`i` be the result of computing :math:`\ine_{\K{128}}(c_1, 0)`.
+
+4. Push the value :math:`\I32.\CONST~i` onto the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~\shape\K{.}\ANYTRUE &\stepto& (\I32\K{.}\CONST~i)
+     & (\iff i = \ine_{\K{128}}(c_1, 0)) \\
+   \end{array}
+
+
+.. _exec-simd-bitmask:
+
+:math:`t\K{x}N\K{.}\BITMASK`
+............................
+
+1. Assert: due to :ref:`validation <valid-vtestop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`i_1^N` be the sequence :math:`\simdto_{t\K{x}N}(c)`.
+
+4. Let :math:`B` be the :ref:`bit width <syntax-valtype>` :math:`|t|` of :ref:`value type <syntax-valtype>` :math:`t`.
+
+5. Let :math:`i_2^N` be the sequence as a result of computing :math:`\ilts_{B}(i_1^N, 0^N)`.
+
+6. Let :math:`c` be the integer :math:`\ibits_{32}^{-1}(i_2^N~0^{32-N})`.
+
+7. Push the value :math:`\I32.\CONST~c` onto the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~\shape\K{.}\BITMASK &\stepto& (\I32\K{.}\CONST~c)
+     & (\iff c = \ibits_{32}^{-1}(\ilts_{B}(\simdto_{t\K{x}N}(c), 0^N)))
+     \\
+   \end{array}
+
+
+.. _exec-simd-narrow:
+
+:math:`t_2\K{x}N\K{.}\NARROW\_t_1\K{x}M\_\sx`
+.............................................
+
+1. Assert: due to :ref:`validation <valid-vtestop>`, two values of :ref:`value type <syntax-valtype>` |V128| are on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_2` from the stack.
+
+3. Let :math:`d_2^M` be the result of computing :math:`\narrow^{\sx}_{t_1, t_2}(\simdto_{t_1\K{x}M}(c_2))`.
+
+4. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+5. Let :math:`d_1^M` be the result of computing :math:`\narrow^{\sx}_{t_1, t_2}(\simdto_{t_1\K{x}M}(c_1))`.
+
+6. Let :math:`c` be the result of :math:`\simdto^{-1}_{t_2\K{x}N}(d_1^M~d_2^M)`.
+
+7. Push the value :math:`\V128.\CONST~c` onto the stack.
+
+.. math::
+   \begin{array}{l}
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~t_2\K{x}N\K{.}\NARROW\_t_1\K{x}M\_\sx &\stepto& (\V128\K{.}\CONST~c)
+   \end{array}
+   \\ \qquad
+     \begin{array}[t]{@{}r@{~}l@{}}
+     (\iff & d_1^M = \narrow^{\sx}_{t_1, t_2}( \simdto_{t_1\K{x}M}(c_1)) \\
+     \wedge & d_2^M = \narrow^{\sx}_{t_1, t_2}( \simdto_{t_1\K{x}M}(c_2)) \\
+     \wedge & c = \simdto^{-1}_{t_2\K{x}N}(d_1^M~d_2^M)
+     \end{array}
+   \end{array}
+
+
+.. _exec-simd-widen:
+
+:math:`t_2\K{x}N\K{.}\WIDEN\_\K{low}\_t_1\K{x}M\_\sx`
+.....................................................
+
+1. Assert: due to :ref:`validation <valid-vtestop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`i_1^N` be the sequence :math:`\simdto_{t_1\K{x}M}(c_1)[0 \slice N]`.
```

Comment:
Maybe just me, but these steps make me uncomfortable unless I know that N = M * 2. Maybe we can assert here? (same for the load extend below).

Reply:
I'm not sure how to assert here, because we always assert due to validation, and this isn't really a validation as N = M * 2 is guaranteed by the accepted syntax.